### PR TITLE
Documentation pass for modular viewport camera controller

### PR DESCRIPTION
**Pull request by hultonha**
This is a follow-on PR from https://github.com/o3de/o3de/pull/2346 to keep the functional and doc changes separate.

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

File: Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/Viewport/ModularViewportCameraController.h [hunk 1]
```diff
@@ -16,37 +16,44 @@
...
         //! Sets up a camera list based on this controller's CameraListBuilderCallback
         void SetupCameras(AzFramework::Cameras& cameras);
         //! Sets up properties shared across all cameras
         void SetupCameraProperies(AzFramework::CameraProps& cameraProps);
 
     private:
-        CameraListBuilder m_cameraListBuilder;
-        CameraPropsBuilder m_cameraPropsBuilder;
+        CameraListBuilder
+            m_cameraListBuilder; //!< Builder to generate a list of CameraInputs to run in the ModularViewportCameraControllerInstance.
```
Comment (amzn-jillich) [inline Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/Viewport/ModularViewportCameraController.h:42]: Nit: Do we need a newline here? Can't we just place the comment above or something?

File: Gems/Atom/Tools/AtomToolsFramework/Code/Source/Viewport/ModularViewportCameraController.cpp [hunk 2]
```diff
@@ -180,32 +181,34 @@ namespace AtomToolsFramework
                     return t * t * t * (t * (t * 6.0f - 15.0f) + 10.0f);
                 };
 
-                const float transitionT = smootherStepFn(m_animationT);
+                const auto& [transformStart, transformEnd, animationT] = m_cameraAnimation;
+
+                const float transitionT = smootherStepFn(animationT);
```
Comment (amzn-jillich) [inline Gems/Atom/Tools/AtomToolsFramework/Code/Source/Viewport/ModularViewportCameraController.cpp:186]: Nit: camera/transitionT -> camera/TransitionTime? and a comment if the time is in seconds as we're doing a check against it with 1.0 below?

**Review by amzn-jillich — APPROVED** [review]

File: Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/Viewport/ModularViewportCameraController.h (hunk 1 above)
Comment (hultonha) [reply Gems/Atom/Tools/AtomToolsFramework/Code/Include/AtomToolsFramework/Viewport/ModularViewportCameraController.h:42]: Sure I can move the comment to the line above (this formatting is performed by `clang-format` to maintain a line limit of 140 characters)

File: Gems/Atom/Tools/AtomToolsFramework/Code/Source/Viewport/ModularViewportCameraController.cpp (hunk 2 above)
Comment (hultonha) [reply Gems/Atom/Tools/AtomToolsFramework/Code/Source/Viewport/ModularViewportCameraController.cpp]: Good shout, have updated 👍

**Review by jonawals — APPROVED** [review]

**Pull request merged**